Content.Server/NPC/Systems/FactionExceptionSystem.cs

```diff
@@ -7,27 +8,115 @@ namespace Content.Server.NPC.Systems;
+
+    private void OnTrackerShutdown(EntityUid uid, FactionExceptionTrackerComponent component, ComponentShutdown args)
+    {
+        foreach (var ent in component.Entities)
+        {
+            if (!_exceptionQuery.TryGetComponent(ent, out var comp))
+                continue;
+            comp.Ignored.Remove(uid);
+            comp.Hostiles.Remove(uid);
+        }
+    }
+
     /// <summary>
     /// Returns whether the entity from an enemy faction won't be attacked
     /// </summary>
-    public bool IsIgnored(FactionExceptionComponent comp, EntityUid target)
+    public bool IsIgnored(EntityUid uid, EntityUid target, FactionExceptionComponent? comp = null)
     {
+        if (!Resolve(uid, ref comp, false))
+            return false;
+
         return comp.Ignored.Contains(target);
     }
 
+    /// <summary>
+    /// Returns the specific hostile entities for a given entity.
+    /// </summary>
+    public IEnumerable<EntityUid> GetHostiles(EntityUid uid, FactionExceptionComponent? comp = null)
+    {
+        if (!Resolve(uid, ref comp, false))
+            return new HashSet<EntityUid>();
+
+        return comp.Hostiles;
+    }
+
     /// <summary>
     /// Prevents an entity from an enemy faction from being attacked
     /// </summary>
-    public void IgnoreEntity(FactionExceptionComponent comp, EntityUid target)
+    public void IgnoreEntity(EntityUid uid, EntityUid target, FactionExceptionComponent? comp = null)
     {
+        comp ??= EnsureComp<FactionExceptionComponent>(uid);
         comp.Ignored.Add(target);
+        EnsureComp<FactionExceptionTrackerComponent>(target).Entities.Add(uid);
     }
 
     /// <summary>
     /// Prevents a list of entities from an enemy faction from being attacked
     /// </summary>
-    public void IgnoreEntities(FactionExceptionComponent comp, IEnumerable<EntityUid> ignored)
+    public void IgnoreEntities(EntityUid uid, IEnumerable<EntityUid> ignored, FactionExceptionComponent? comp = null)
+    {
+        comp ??= EnsureComp<FactionExceptionComponent>(uid);
+        foreach (var ignore in ignored)
+        {
+            IgnoreEntity(uid, ignore, comp);
+        }
+    }
+
+    /// <summary>
+    /// Makes an entity always be considered hostile.
+    /// </summary>
+    public void AggroEntity(EntityUid uid, EntityUid target, FactionExceptionComponent? comp = null)
+    {
+        comp ??= EnsureComp<FactionExceptionComponent>(uid);
+        comp.Hostiles.Add(target);
+        EnsureComp<FactionExceptionTrackerComponent>(target).Entities.Add(uid);
+    }
+
+    /// <summary>
+    /// Makes an entity always be considered hostile.
+    /// </summary>
+    public void DeAggroEntity(EntityUid uid, EntityUid target, FactionExceptionComponent? comp = null)
+    {
+        comp ??= EnsureComp<FactionExceptionComponent>(uid);
+        if (!comp.Hostiles.Remove(target) || !_trackerQuery.TryGetComponent(target, out var tracker))
+            return;
+        tracker.Entities.Remove(uid);
+    }
```

Comment:
This can cause an empty FactionExceptionComponent so ideally you'd just return if a resolve fails.